packages/runtime/src/templates/server.ts

```diff
@@ -0,0 +1,73 @@
+import https from 'https'
+
+import { NodeRequestHandler, Options } from 'next/dist/server/next-server'
+
+import { getNextServer, NextServerType } from './handlerUtils'
+
+const NextServer: NextServerType = getNextServer()
+
+interface NetlifyNextServerOptions extends Options {
+  netlifyRevalidateToken: string
+}
+
+class NetlifyNextServer extends NextServer {
```

Comment:
This is looking good @orinokai. I haven't tried it, but I wonder if it makes more sense to use a [Proxy](https://developer.mozilla.org/en-US/docs/web/javascript/reference/global_objects/proxy) instead of extending the NextServer. 🤔

Reply:
I initially attempted to implement this with dependency injection, rather than inheritance, but the `NextServer` recursively calls `this.getRequestHandler` during the `res.revalidate` call and I need the nested call to still hit our overridden version. I couldn't work out how to do that with DI and I think it might be the same using a Proxy?